src/plugins/template/src/template_plugin.cpp

```diff
@@ -95,64 +75,94 @@ void TransformNetwork(std::shared_ptr<ngraph::Function>& function,
 
     // After `run_passes`, we have the transformed function, where operations match device operations,
     // and we can create device backend-dependent graph
-    passManager.run_passes(function);
+    passManager.run_passes(model);
 }
 // ! [plugin:transform_network]
 
 // ! [plugin:load_exe_network_impl]
-InferenceEngine::IExecutableNetworkInternal::Ptr Plugin::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork& network,
-                                                                            const ConfigMap& config) {
-    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "Plugin::LoadExeNetworkImpl");
-
-    InferenceEngine::InputsDataMap networkInputs = network.getInputsInfo();
-    InferenceEngine::OutputsDataMap networkOutputs = network.getOutputsInfo();
-
-    auto fullConfig = Configuration{config, _cfg};
-    return std::make_shared<ExecutableNetwork>(network.getFunction(),
-                                               networkInputs,
-                                               networkOutputs,
-                                               fullConfig,
-                                               std::static_pointer_cast<Plugin>(shared_from_this()));
+std::shared_ptr<ov::ICompiledModel> TemplatePlugin::Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                                          const ov::AnyMap& properties) const {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "Plugin::compile_model");
+
+    auto fullConfig = Configuration{properties, _cfg};
+    auto streamsExecutorConfig =
+        InferenceEngine::IStreamsExecutor::Config::MakeDefaultMultiThreaded(fullConfig._streamsExecutorConfig);
+    streamsExecutorConfig._name = "TemplateStreamsExecutor";
```

Comment:
we have the same name in the class dtor and import_model. Maybe we need to store it as a constant somewhere?